src/main/scala/co/ledger/wallet/daemon/clients/ScalaHttpClient.scala

```diff
@@ -57,40 +59,9 @@ class ScalaHttpClient(implicit val ec: ExecutionContext) extends co.ledger.core.
     !(statusCode >= 200 && statusCode < 400)
   }
 
-  private def readRequestBody(request: HttpRequest, os: OutputStream) = {
-    val dataOs = new DataOutputStream(os)
-    try {
-      dataOs.write(request.getBody)
-      dataOs.flush()
-    }
-    finally dataOs.close()
-  }
-
-  private def setRequestProperties(request: HttpRequest, connection: HttpURLConnection) = {
-    connection.setRequestMethod(resolveMethod(request.getMethod))
-    for ((key, value) <- request.getHeaders.asScala) {
-      connection.setRequestProperty(key, value)
-    }
-    connection.setRequestProperty("User-Agent", "ledger-lib-core")
-    connection.setRequestProperty("Content-Type", "application/json")
-  }
-
-  private def createConnection(request: HttpRequest) = {
-    DaemonConfiguration.proxy match {
-      case None => new URL(request.getUrl).openConnection().asInstanceOf[HttpURLConnection]
-      case Some(proxy) =>
-        new URL(request.getUrl)
-          .openConnection(
-            new java.net.Proxy(
-              java.net.Proxy.Type.HTTP,
-              new InetSocketAddress(proxy.host, proxy.port)))
-          .asInstanceOf[HttpURLConnection]
-    }
-  }
-
-  private def readResponseBody(connection: HttpURLConnection, onError: Boolean): HttpReadBodyResult = {
+  private def readResponseBody(resp: Response, onError: Boolean): HttpReadBodyResult = {
     val response =
-      new BufferedInputStream(if (!onError) connection.getInputStream else connection.getErrorStream)
+      new BufferedInputStream(if (!onError) resp.getInputStream else new ByteArrayInputStream(new Array[Byte](0)))
```

Comment:
Original code `connection.getErrorStream`, in Java doc, it's returning the connection errors, not errors of HTTP.
So I think the origin code is wrong.
From the purpose of this function, I think it should be just:
```suggestion
      new BufferedInputStream(connection.getInputStream)
```
What do you think?

Reply:
Legacy code too, let's test this :D